docker-compose.build.yml

```diff
@@ -0,0 +1,17 @@
+---
+version: "2.4"
+
+services:
+  build:
+    image: golang:1.18.0
+    volumes:
+      - .:/go/src/github.com/balena-io/sshproxy
+    working_dir: /go/src/github.com/balena-io/sshproxy
+    entrypoint:
+      - /bin/bash
+      - "-c"
+    command:
+      - |
+        set -axe
+
+        make -j $$(nproc) dep release
```

Comment:
Rather than having two compose files, we might be able to replace the build command with
```
docker compose -f docker-compose.test.yml run sut make -j $(nproc) dep release
```